### PR TITLE
Fix regression in supporting `user:port@host` app db connection strings

### LINKED ISSUE #20121: Fix `MB_DB_CONNECTION_URI` with format like `postgres://dan:password@localhost:5432/metabase`
Regression in `master` (43)
Regression of #14678

`postgres://dan:password@localhost:5432/metabase` isn't a valid JDBC connection string. Postgres wants `postgresql://localhost:5432/metabase?user=dan&password=password`. This used to work prior to #19970 when we parsed `MB_DB_CONNECTION_URI` but stopped working now that we no longer parse it (we do still however correct `postgres:` to `postgresql:`).

We should detect this one specific case and fix it with as little parsing as humanly possible. Since `java.sql.DriverManager` supports passing in `user` and `password` separately from the rest of the connection string (either as a pair of strings or inside a `java.util.Properties`) we should make sure we do that so we don't have to tack on `?user=` or `&user=`.

### PR DESCRIPTION
**Pull request by camsaul**
Fixes #20121

### CODE REVIEW

File: test/metabase/db/data_source_test.clj [hunk 1]
```diff
@@ -1,18 +1,53 @@
...
-  (let [data-source (mdb.data-source/broken-out-details->DataSource
-                     :h2
-                     {:subprotocol "h2"
-                      :db          (format "mem:%s" (mt/random-name))
-                      :classname   "org.h2.Driver"})]
-    (with-open [conn (.getConnection data-source)]
-      (is (= [{:one 1}]
-             (jdbc/query {:connection conn} "SELECT 1 AS one;"))))))
+  (testing :postgres
+    (is (= (->DataSource
```
Comment (dpsutton) [inline test/metabase/db/data_source_test.clj:15]: This really confused me at first until I realized you put your own notion of equality in it. But I see that you are comparing properties and the url. A problem with this is that url equality works as:

```
 Two URL objects are equal if they have the same protocol, reference equivalent hosts, have the same port number on the host, and the same file and fragment of the file.

Two hosts are considered equivalent if both host names can be resolved into the same IP addresses; else if either host name can't be resolved, the host names must be equal without regard to case; or both host names equal to null.
```
Which means we are making network requests for this comparison. That's always a weird footgun with this. I'm not sure if that's something we should live with in testing or be wary of.

Comment (dpsutton) [reply test/metabase/db/data_source_test.clj:15]: Ah. I totally misread this. Its just string comparison. Ignore me

**Review by dpsutton — APPROVED** [review]

**Comment by dpsutton** [comment]
Actually, one passing thought. Are we sure it is always `user` vs `username`? Is that consistent across all dbs?

File: test/metabase/db/data_source_test.clj (hunk 1 above)
Comment (camsaul) [reply test/metabase/db/data_source_test.clj:15]: yup. No `URL` or `URI` involved. It's just a string and the `Properties` map which works like you'd expect

**Comment by camsaul** [comment]
@dpsutton well it's definitely `user` for Postgres, and MySQL/MariaDB, the only two we support setting `MB_DB_USER` for for the application DB. Might not be for SQL Server or whatever but we don't support SQL Server as the application DB 🤷

**Pull request merged**